src/Fizzics/level2.js

```diff
@@ -30,6 +30,13 @@ var FizzicsLevel2 = GObject.registerClass({
             this._compile();
         });
         this._content.add(this._codeview);
+
+        this.connect('destroy', () => {
+            if (this._model && this._notifyHandler) {
+                this._model.disconnect(this._notifyHandler);
+                this._notifyHandler = null;
```

Comment:
We should also check if adding an `unbindModel` (empty) method to the base class, and calling it in there in its `shutdown` method. So instead of each widget having to remember to connect to the destroy signal, they'd just have to disconnect whatever they connected during the `bindModel`, if that makes sense (and works).

Reply:
> Not that it matters a lot here, but the signal handlers are integers, so this should check if they're `0` and reset it as such IMO. Same comment for all the rest of the uses.

As far as I know, the handler can't be 0, this is the error that I got when I use an invalid handler_id:
```
g_signal_handler_disconnect: assertion 'handler_id > 0' failed
```

Reply:
Of course. The handler is an unsigned long ( https://developer.gnome.org/gobject/stable/gobject-Signals.html#g-signal-handler-disconnect ).

My point is that you're treating the handler here like an object (checking/reassigning to `null`) but it's a ulong, so it should be checked and reset as `0`.

Reply:
Okay, now I undertstand this and I've changed it.

Reply:
JS isn't typed, so it doesn't matter that the handler ID is otherwise an integer. Using `null` for a not-yet-assigned class member is a common idiom and the reason why we do it is so that the programmer only has to remember to check `!== null` or `=== null` instead of worrying about any of JS's weird type coercions that may apply.